views/js/apipCreator/api/accessElement.js

```diff
@@ -0,0 +1,233 @@
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ *
+ */
+define([
+    'lodash',
+    'taoQtiItem/apipCreator/api/authoringObject',
+    'taoQtiItem/apipCreator/api/qtiElement',
+    'taoQtiItem/apipCreator/api/accessElementInfo'
+], function (_, authoringObject, QtiElement, AccessElementInfo) {
+    'use strict';
+    
+    /**
+     * accessElement constructor
+     * @param {object} apipItem apipItem creator api instance 
+     * @see {@link package-tao\taoQtiItem\views\js\apipCreator\api\apipItem.js}
+     * @param {type} node XML element node. If not given then new empty node will be created and appended to the item XML.
+     * @returns {accessElement_L19.AccessElement}
+     */
+    function AccessElement(apipItem, node) {
+        if (node === undefined) {
+            node = this.createXMLNode(apipItem);
+        }
+        authoringObject.init(this, apipItem, node);
+    }
+
+    /**
+     * Create accessElement xml node and append it to the item XML
+     * @param {object} apipItem apipItem creator api instance 
+     * @see {@link package-tao\taoQtiItem\views\js\apipCreator\api\apipItem.js}
+     * @returns {Object} created XML node
+     */
+    AccessElement.prototype.createXMLNode = function createXMLNode(apipItem) {
+        var accessElementNum = apipItem.xpath('//apip:accessElement').length,
+            accessElementNode = apipItem.createNode(
+                'apip',
+                'accessElement',
+                {"identifier" : ('ae' + accessElementNum)}
```

Comment:
are you sure that this would be unique ? what we have imported an apip item from thrid party-vendors or, if we remove and delete an accessElement [ae0, ae1, ae2], we remove ae1, then we create a new one [ae0, ae2, ae2] 

This reminds me this function : https://github.com/oat-sa/extension-tao-itemqti/blob/master/views/js/qtiItem/core/IdentifiedElement.js#L3-L37
Maybe you may reuse the same idea ? or anything else ?